projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts

```diff
@@ -24,20 +24,20 @@ export abstract class AbstractMultichoiceAutocompleteFieldComponentComponent imp
     filteredOptions: Observable<Array<MultichoiceFieldValue>>;
 
     ngOnInit() {
-        this.filteredOptions = this.formControlRef.valueChanges.pipe(
-            startWith(''),
-            map(value => this._filter(value).filter((option) => !this.multichoiceField.value.includes(option.key)))
-        );
+        this.formControlRef.valueChanges.subscribe(newVal => {
+            this.filteredOptions = of(this._filter(newVal ?? '').filter((option) => !this.multichoiceField.value?.includes(option.key)));
+        })
     }
 
     ngOnDestroy(): void {
         this.filteredOptions = undefined;
     }
 
     add(event: MatChipInputEvent): void {
-        const value = (event['key'] || '').trim();
+        const value = (event['key'] || '');
```

Comment:
Are the keys only strings? If the keys were numbers then `0` would evaluate to `''` which is bad. Maybe `??` would be better than `||`.

Reply:
Done